source/grammar/index.rst

```diff
@@ -28,50 +28,58 @@ repetition is denoted by curly brackets ``+`` (one or more) and ``*`` (zero or m
         :| statement
     global_statement: subroutine_declaration
         :| kernel_declaration
-        :| gate_definition
+        :| quantum_gate_definition
         :| calibration
     statement: expression_statement
         :| declaration_statement
         :| selection_statement
         :| iteration_statement
         :| selection_directive_statement
+        :| alias_statement
         :| quantum_statement
         :| timing_box
         :| pragma
-    subroutine_declaration: "def" id args_declaration return_signature? program_block
-    args_declaration : "(" type_and_id_list ")"
-    type_and_id_list: ( classical_type_and_id_list | quantum_type_and_id_list )*
+        :| comment
+    comment: "//" ... | "/*" ... "*/"
+    subroutine_declaration: "def" id subroutine_args return_signature? program_block
+    subroutine_args: classical_params | quantum_type_and_id_list
     classical_type_and_id_list: ( declare_type association "," )* declare_type association
-    declare_type: dependent_type_declaration 
-        :| independent_type_specifier
-    dependent_type_declaration: dependent_type_specifier designator
-    dependent_type_specifier: "int" | "uint" | "angle" | "fixed" | "stretch"
+    bit_type: "bit" | "creg"
+    bit_declaration: bit_type id designator
+    declare_type: dependent_type_specifier | independent_type_specifier | bit_type
+    dependent_type_declaration: (dependent_type_specifier designator | "fixed" double_designator) id
+    dependent_type_specifier: "int" | "uint" | "angle" | "stretch"
     designator: "[" expression "]"
-    independent_type_specifier: "bit" | "creg" | "bool" 
-        :| timing_type
+    double_designator: "[" expression "," expression "]"
+    independent_type_specifier: "bool" | timing_type
+    independent_type_declaration: independent_type_specifier id
     association: ":" id
     return_signature: "->" classical_declaration
-    constant_declaration: "const" ( dependent_type_declaration | independent_type ) assignment 
-    classical_declaration: ( dependent_type_declaration | independent_type ) assignment?
+    constant_declaration: "const" id assignment_expression
+    classical_declaration: ( bit_declaration | dependent_type_declaration
+        :| independent_type_declaration ) assignment_expression?
     program_block: "{" ( program_block | statement ) "}"
     kernel_declaration: "kernel" id classical_type_and_id_list return_signature? ";"
+    alias_statement: "let" id "=" concatenate_expression
+    concatenate_expression: "=" ( id range | id "||" id | id "[" expression_list "]"
```

Comment:
```suggestion
    alias_statement: "let" id "=" expression
    concatenate_expression: "=" ( id range | id "||" id | id "[" expression_list "]"
```

Reply:
I could do this, but I think it would allow many options which are not allowed. The `alias` is just for registers and has these operations defined. I wouldn't think we would want to include something like a function call?

But maybe this is the grammar being too specific.

Reply:
I think too specific.